File: backend/src/main/java/com/ufc/backend/backend/exceptions/ClassDontHaveThePreRequisiteException.java
package com.ufc.backend.backend.exceptions;

import com.ufc.backend.backend.model.Classes;

public class ClassDontHaveThePreRequisiteException extends RuntimeException {
    public ClassDontHaveThePreRequisiteException(Classes classes) {
        super("Você não pode cursar essa cadeira: " + classes.getName() + ", Você precisa cursar o pré requisito dela antes: " + classes.getPreRequisite().getName());
    }

    public ClassDontHaveThePreRequisiteException(String msg, Throwable cause) {
        super(msg, cause);
    }
}

File: backend/src/main/java/com/ufc/backend/backend/exceptions/ClassDontHaveThePreRequisiteException.kt
package com.ufc.backend.backend.exceptions

import com.ufc.backend.backend.model.Subject

class ClassDontHaveThePreRequisiteException(subject: Subject) : RuntimeException(
    "Você não pode cursar essa cadeira: ${subject.name}, você precisa cursar o pré requisito dela antes: ${subject.preRequisite?.name}"
)
